### PR TITLE
[SES-2162] - Remove wrapping of config message

### PR DESCRIPTION
**Pull request by simophin**
**Background**

_tldr; Android is doing incorrect config message encoding._

Sometime in the past, any user's config update (like what groups they subscribed to, contacts, etc) will be pushed into the swarm as "messages", just like any other messages the user may see, but with a different type. Because they all push into the same place, when the user tries to restore their account, as all the messages are stored and restored in date sequential order, the user might not get the "config message" in time, if not at all.

To fix the problem, we introduced the concept of `namespaces`, essentially pre-defined constants with meaning. Like `GROUPS` namespace dictates that the messages should be group config messages.  By doing this important messages like config messages won't get swamped along side other normal user messages, this improve the reliability greatly.

As a side effect coming out from the namespace changes, we can know in advance what the message format is for a given namespace. This eliminates the need to use `protobuf`: where different  types of messages are serialised together.

So we did move away from the protobuf for the config messages (regular messages still use protobuf!). However, Android codebase still keeps using protobuf for the config messages, this causes trouble for the other clients who no longer use protobuf, as those clients won't be able to decode the protobuf-wrapped messages.

**This PR:**

1. Removes the wrapping/unwrapping of the config messages on the client side, any config message serialisation is now handed over entirely to `libsession`
2. Update `libsession` so that it can handle already-wrapped messages that exist in the wild, so that existing users' configs are preserved.

**Note**
This PR is a temporarily fix for the current production. The config system has been reworked on the groups branch and once that one is about to be merged, all changes here can be discarded first.

### CODE REVIEW

**Review by mpretty-cyro — APPROVED** [review]

**Review by AL-Session — APPROVED** [review]
Looks great! Nice one!

File: libsession/src/main/java/org/session/libsession/messaging/sending_receiving/pollers/Poller.kt [hunk 1]
```diff
@@ -126,37 +127,26 @@ class Poller(private val configFactory: ConfigFactoryProtocol, debounceTimer: Ti
...
-        val messages = SnodeAPI.parseRawMessagesResponse(
-            rawMessages,
-            snode,
-            userPublicKey,
-            namespace,
-            updateLatestHash = true,
-            updateStoredHashes = true,
-        )
+        val messages = rawMessages["messages"] as? List<*>
+        val processed = if (messages != null && messages.isNotEmpty()) {
```
Comment (bemusementpark) [inline libsession/src/main/java/org/session/libsession/messaging/sending_receiving/pollers/Poller.kt:131]:
```suggestion
        val processed = if (messages.isNotNullOrEmpty()) {
```

Comment (simophin) [reply libsession/src/main/java/org/session/libsession/messaging/sending_receiving/pollers/Poller.kt]: Updated

**Review by ThomasSession — APPROVED** [review]

File: libsession/src/main/java/org/session/libsession/messaging/sending_receiving/pollers/Poller.kt [hunk 2]
```diff
@@ -126,37 +127,26 @@ class Poller(private val configFactory: ConfigFactoryProtocol, debounceTimer: Ti
...
-                    Log.w("Loki", "shared config message handled in configs wasn't SharedConfigurationMessage but was ${message.javaClass.simpleName}")
-                    return@forEach
-                }
-                val merged = forConfigObject.merge(hash!! to message.data).firstOrNull { it == hash }
-                if (merged != null) {
-                    // We successfully merged the hash, we can now update the timestamp
-                    latestMessageTimestamp = if ((message.sentTimestamp ?: 0L) > (latestMessageTimestamp ?: 0L)) { message.sentTimestamp } else { latestMessageTimestamp }
-                }
+                forConfigObject.merge(hash to body)
+                latestMessageTimestamp = if (timestamp > (latestMessageTimestamp ?: 0L)) { timestamp } else { latestMessageTimestamp }
```
Comment (bemusementpark) [inline libsession/src/main/java/org/session/libsession/messaging/sending_receiving/pollers/Poller.kt:149]:
```suggestion
                latestMessageTimestamp = max(timestamp, latestMessageTimestamp ?: 0L)
```

Comment (ThomasSession) [reply libsession/src/main/java/org/session/libsession/messaging/sending_receiving/pollers/Poller.kt:149]: This changes the behaviour as it would never return a null `latestMessageTimestamp`. That nullability is used as a check further down and would be set to 0L in this case and pass instead of being caught by that if.

Comment (simophin) [reply libsession/src/main/java/org/session/libsession/messaging/sending_receiving/pollers/Poller.kt:149]: The changes here are coming from the groups branch and they will be reverted before the groups are merged back in, I'm not sure if we want to spend too much energy on these
